### DISCUSSION TITLE
k8s多机集群环境启动session遇到的问题

### DISCUSSION THREAD

**Discussion by nanyoullm** [alibaba/GraphScope#1598] [Q&A] [answered]
请教个问题，目前已裸机搭建了k8s集群（一台master，两台nodes），
![image](https://user-images.githubusercontent.com/13297712/169729682-22176bfe-c481-4cd8-a5e3-c5dec2b8aa1c.png)
随后在master中拉起session：import graphscope; sess=graphscope.session();
报错内容：
![image](https://user-images.githubusercontent.com/13297712/169729744-fde07f32-16ea-406f-96af-cfe6785d025e.png)
![image](https://user-images.githubusercontent.com/13297712/169729763-c0a7f341-90e3-4371-b72a-5c28d979c6fd.png)
![image](https://user-images.githubusercontent.com/13297712/169729777-5880604f-93f1-4914-aeb4-2498439738e8.png)

请问是什么原因呢，是网络连接问题吗？虚心求教！

**Comment by nanyoullm** [comment]
@acezen 求教😁

**Reply by nanyoullm** [reply]
报错位置：
~/anaconda3/lib/python3.9/site-packages/graphscope/deploy/kubernetes/cluster.py
line 469：
        if wait_for_deployment_complete(
            api_client=self._api_client,
            namespace=self._namespace,
            name=self._coordinator_name,
            timeout_seconds=self._saved_locals["timeout_seconds"],
        )
卡住超时，应该是集群的问题

 ~/anaconda3/lib/python3.9/site-packages/graphscope/deploy/kubernetes/utils.py
line 114:
        if (
            s.updated_replicas == response.spec.replicas
            and s.replicas == response.spec.replicas
            and s.available_replicas == response.spec.replicas
            and s.observed_generation >= response.metadata.generation
        )
中， s.available_replicas  == response.spec.replicas 为False，s.available_replicas=None， response.spec.replicas=1

 ~/anaconda3/lib/python3.9/site-packages/graphscope/deploy/kubernetes/utils.py
line 133:
if (
                        not container_status.ready
                        and container_status.restart_count > 0
                    )
container_status.ready 为 False

**Comment by lidongze0629** [comment]
@nanyoullm 报错来看是 client 所在的机器和 K8s ApiServer 通信失败了。方便告诉下 k8s 集群的部署方法嘛(例如是不是虚拟机部署)。同时，你也在拉起的过程中，通过 `kubectl -n <namespace> describe pod <pod_name>` 来看一下 `coordinator pod`的状态。

**Reply by nanyoullm** [reply]
不是虚拟机部署哈，是三台云机器（一台master，两台node），刚刚又重新部署了一边集群，还在尝试中，`coordinator pod`的状态的也尝试看看~

**Reply by nanyoullm** [reply]
> @nanyoullm 报错来看是 client 所在的机器和 K8s ApiServer 通信失败了。方便告诉下 k8s 集群的部署方法嘛(例如是不是虚拟机部署)。同时，你也在拉起的过程中，通过 `kubectl -n <namespace> describe pod <pod_name>` 来看一下 `coordinator pod`的状态。

![image](https://user-images.githubusercontent.com/13297712/169954908-1980d28c-f3d0-4b94-a95d-9086d03c859e.png)
![image](https://user-images.githubusercontent.com/13297712/169954831-ed028de5-418e-4a5e-a889-fdb9b5223170.png)

**Comment by nanyoullm** [comment, marked as answer]
已启动成功，是k8s容器网络配置的问题，我下载calico.yaml并部署后，可以拉起session
谢谢 @lidongze0629 的提醒
![image](https://user-images.githubusercontent.com/13297712/169980112-8934d0af-e1ff-4bd6-a720-0bfd7a89ec6a.png)
